### ISSUE TITLE
bad function formatting with no parameter

### ISSUE THREAD

**Issue by scampi** [rust-lang/rustfmt#3278]
The snippet below is badly formatted, where the empty parameters list spends over 2 lines:

```rust
pub fn parse_conditional<'a, I: 'a>() -> impl Parser<Input = I, Output = Expr, PartialState = ()> + 'a
where I: Stream<Item = char>
{
    // do something
}
```

```rust
pub fn parse_conditional<'a, I: 'a>(
) -> impl Parser<Input = I, Output = Expr, PartialState = ()> + 'a
where
    I: Stream<Item = char>,
{
    // do something
}
```

----

This happens with the latest from master 996d5e8d6ce2e756f50bfa51b9622c50f725415c

**Comment by scampi** [comment]
With the given example, only ` 'a` is over the threshold. A formatting would be possible ?

```rust
pub
fn parse_conditional<'a, I: 'a>() -> impl Parser<Input = I, Output = Expr, PartialState = ()> + 'a
where
    I: Stream<Item = char>,
{
    // do something
}
```

If for another case the previous didn't succeed, maybe the following ?

```rust
pub fn parse_conditional<'a, I: 'a>()
-> impl Parser<Input = I, Output = Expr, PartialState = ()> + 'a
where
    I: Stream<Item = char>,
{
    // do something
}
```

**Comment by topecongiro** [comment]
Closed via #3294.
